src/shared/components/oncoprint/OncoprintUtils.ts

```diff
@@ -531,6 +538,208 @@ export function getCategoricalTrackRuleSetParams(
+            if (oqlFiltered) {
+                const geneSymbol = query.hugoGeneSymbol.toUpperCase();
+                mutations = oqlFiltered.filter(
+                    (alt: any) =>
+                        alt.hugoGeneSymbol &&
+                        alt.hugoGeneSymbol.toUpperCase() === geneSymbol &&
+                        alt.molecularProfileAlterationType ===
+                            'MUTATION_EXTENDED'
+                );
+            }
+        } else {
+            const mutationPromise = oncoprint.props.store.annotatedMutationCache.get(
+                {
+                    entrezGeneId: query.entrezGeneId,
+                }
+            );
+            mutations = mutationPromise.result;
+            if (mutations) {
+                mutations = mutations.filter(
+                    m => m.molecularProfileId === query.molecularProfileId
+                );
+            }
+        }
+
+        if (mutations && mutations.length > 0) {
+            const sampleMutationMap = new Map<string, number | null>();
+
+            mutations.forEach(m => {
+                const vafReport = getVariantAlleleFrequency(m);
+                if (
+                    vafReport &&
+                    typeof vafReport.vaf === 'number' &&
+                    !isNaN(vafReport.vaf)
+                ) {
+                    const existingVaf = sampleMutationMap.get(
+                        m.uniqueSampleKey
+                    );
+                    if (
+                        existingVaf === undefined ||
+                        vafReport.vaf > existingVaf!
+                    ) {
+                        sampleMutationMap.set(m.uniqueSampleKey, vafReport.vaf);
+                    }
+                } else {
+                    if (!sampleMutationMap.has(m.uniqueSampleKey)) {
+                        sampleMutationMap.set(m.uniqueSampleKey, null);
+                    }
+                }
+            });
+
+            // Filter to only include profiled samples
+            // Skip filter if gene is not in query
+            return samples
+                .filter(sample => {
+                    if (!isGeneInQuery) {
+                        return true;
+                    }
+                    return isSampleProfiled(
+                        sample.uniqueSampleKey,
+                        query.molecularProfileId,
+                        query.hugoGeneSymbol,
+                        coverageInfo
+                    );
+                })
+                .map(sample => {
+                    const vafValue = sampleMutationMap.get(
+                        sample.uniqueSampleKey
+                    );
+                    return {
+                        uniqueSampleKey: sample.uniqueSampleKey,
+                        uniquePatientKey: sample.uniquePatientKey,
+                        value: vafValue !== undefined ? vafValue : null, // null for profiled but not mutated
```

Comment:
we should probably use an explicit constant value like `PROFILED_BUT_NOT_MUTATED` instead of `null`